RingPublishingGDPR/src/main/java/com/ringpublishing/gdpr/RingPublishingGDPR.java

```diff
@@ -97,66 +109,85 @@ public void initialize(@NonNull final Application application,
         this.formViewController = new FormViewController(api, ringPublishingGDPRUIConfig);
         this.activityLifecycleObserver = new ActivityLifecycleObserver(application);
         this.ringPublishingGDPRApplicationCallback = createRingPublishingGDPRApplicationCallback();
         initialized = true;
+        tenantConfiguration.setState(TenantState.LOADING);
 
-        storage.configureGDPRApplies(gdprApplies);
+        fetchTenantConfiguration();
 
-        if (storage.didAskUserForConsents() && !isOutdated() && gdprApplies)
+        if (isOutdated())
         {
-            verifyConsents(context);
+            verifyState = VerifyState.OUTDATED;
         }
+        else
+        {
+            if (storage.didAskUserForConsents())
+            {
+                verifyOutdated();
+            }
+            else
+            {
+                verifyState = VerifyState.ACTUAL;
+            }
+        }
+    }
+
+    void fetchTenantConfiguration()
+    {
+        api.configuration(new ConfigurationCallback()
+        {
+            @Override
+            public void onSuccess(String url, boolean gdprApplies)
+            {
+                setTenantConfiguration(true, url, gdprApplies);
+            }
+
+            @Override
+            public void onFailure()
+            {
+                setTenantConfiguration(false, null, false);
+                Log.w(TAG, "Failure onConfigurationFailure");
+            }
+        });
+    }
+
+    private void setTenantConfiguration(boolean success, String url, boolean gdprApplies)
+    {
+        storage.configureGDPRApplies(gdprApplies);
+        tenantConfiguration.setState(success ? TenantState.SUCCESS : TenantState.FAILURE);
+        tenantConfiguration.setHost(url);
+        tenantConfiguration.setGdprApplies(gdprApplies);
+        formViewController.setTenantConfiguration(tenantConfiguration);
+        apiCallFinished();
     }
 
     /**
      * Use this method to check that RingPublishingGDPRActivity should be displayed
-     * @return true when Consent View should be displayed. It could be case when Consent View was never displayed, or is outdated.
-     * When gdpr not apply for your context, this method will return false
+     * Call this method in first application Activity
+     * Wait for callback in RingPublishingGDPRShowConsentScreenListener to decide that consent screen should be displayed.
      */
-    public boolean shouldShowConsentForm()
+    public void setRingPublishingGDPRShowConsentScreenListener(RingPublishingGDPRShowConsentScreenListener ringPublishingGDPRShowConsentListener)
     {
-        if(!gdprApplies)
+        this.ringPublishingGDPRShowConsentListener = ringPublishingGDPRShowConsentListener;
+
+        if (ringPublishingGDPRShowConsentListener == null)
         {
-            return false;
+            return;
         }
-        return (!storage.didAskUserForConsents() || isOutdated());
+
+        apiCallFinished();
```

Comment:
nie rozumiem dlaczego to tu jest wołane

Reply:
rozumiem, że to metoda synchronizująca 2 niezależne szczały do API... Ale...

Reply:
Tak to synchronizuje 2 strzały do api. 
Można by to zrobić używając RxJava i merge ale nie dokładamy dodatkowych bibliotek do modułu.